osf/models/user.py

```diff
@@ -1508,6 +1508,56 @@ def get_node_comment_timestamps(self, target_id):
         default_timestamp = dt.datetime(1970, 1, 1, 12, 0, 0, tzinfo=pytz.utc)
         return self.comments_viewed_timestamp.get(target_id, default_timestamp)
 
+    def gdpr_delete(self):
+        """
+        This function does not remove the user object reference from our database, but it does disable the account and
+        remove identifying in a manner compliant with GDPR guidelines.
+        :return:
+        """
+        from osf.models import PreprintService, AbstractNode
+
+        user_nodes = self.nodes.all()
+        #  Validates the user isn't trying to delete things they deliberately made public.
+        if user_nodes.filter(type='osf.registration').exists():
+            raise ValidationError('You cannot delete this user because they have one or more registrations.')
+
+        if PreprintService.objects.filter(node___contributors=self, ever_public=True).exists():
+            raise ValidationError('You cannot delete this user because they have one or more preprints.')
+
+        # Validates that the user isn't trying to delete things nodes they are the only admin on.
+        personal_nodes = [node.id for node in user_nodes if node.contributor_set.count() == 1]
```

Comment:
This is the best I could up with, it's hard to count the length of a queryset collect via through table. So I was unable to to this via annotation, not sure if it would be possible or how best to do it, but this works and doesn't really need to be optimized.

Reply:
Does this work?

```python
AbstractNode.objects.annotate(contrib_count=Count('_contributors')).filter(contrib_count__lte=1).filter(contributor__user=me)
```

You'll probably need to add more filtering (filtering out bookmarks, quickfiles etc.) but you get the idea.

Reply:
That will actually work fine verbatim, we do want to remove Quickfiles ect, that's part of the requirements.